src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -27,25 +31,65 @@ namespace CoreNodeModels.Input
     [OutPortDescriptions("Selected Color.")]
     public class ColorPalette : NodeModel
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private DSColor prevColor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
```

Comment:
@mjkkirschner Thanks for the explaination. Generally for boolean, we come up with a name starting with `is` `has` `can` `should`, so in this case maybe `ShouldRecordForUndo`?